src/subscription_manager/cli_command/register.py

```diff
@@ -387,6 +388,9 @@ def _prompt_for_environment(self):
         """
         By breaking this code out, we can write cleaner tests
         """
+        if not self._is_atty():
+            raise system_exit(os.EX_USAGE, "Error: --environments parameter not provided!")
```

Comment:
regarding the text of the message: exclamation marks usually work for games, not in general messages; i'd also be more explicit on the actual reason why sub-man is erroring out, eg something like "--environments is required and it was not specified"